x-pack/plugins/apm/server/routes/service_map/route.ts

```diff
@@ -125,10 +126,7 @@ const serviceMapRoute = createApmServerRoute({
       uiSettingsClient.get<number>(apmServiceGroupMaxNumberOfServices),
     ]);
 
-    const serviceNames = [
-      ...(serviceName ? [serviceName] : []),
-      ...(serviceGroup?.serviceNames ?? []),
-    ];
+    const serviceNames = compact([serviceName]);
 
```

Comment:
What happens if this is an empty array? Does it return a service map for all services or no service map at all?

Reply:
It returns a service map for all services. I think that was there before to handle service-specific service maps